### ISSUE TITLE
styler fails when adding the result of a subsetting expression to a function call

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/styler#573]
**Edit:**

styler fails on the addition of the result of a subsetting expression with a function call, like here: 
``` r
styler::style_text("x[1] + c()")
#> Error in if (!is.null(gg_call) && gg_call$text[gg_call$token == "SYMBOL_FUNCTION_CALL"] == : missing value where TRUE/FALSE needed
```

<sup>Created on 2020-01-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

***

**Initial Post:**

I am using `styler:::style_active_file()` to style an `.Rmd` file, but running into the following error-

```r
> styler:::style_active_file()
Using style transformers `styler::tidyverse_style()`
Error in if (!is.null(gg_call) && gg_call$text[gg_call$token == "SYMBOL_FUNCTION_CALL"] ==  : 
  missing value where TRUE/FALSE needed
```

Maybe the trace gives some insight into this?

```r
> traceback()
24: fun(x)
23: f(init, x[[i]])
22: Reduce(function(x, fun) fun(x), funs, init = pd_flat)
21: visit_one(pd_nested, funs)
20: .f(.x[[i]], ...)
19: map(pd_nested$child, post_visit, funs = funs)
18: .f(.x[[i]], ...)
17: map(pd_nested$child, post_visit, funs = funs)
16: .f(.x[[i]], ...)
15: map(pd_nested$child, post_visit, funs = funs)
14: post_visit(pd_nested, c(transformers$initialize, transformers$line_break, 
        set_multi_line, update_newlines))
13: apply_transformers(pd_nested, transformers)
12: parse_transform_serialize_r(., transformers, warn_empty = warn_empty)
11: function_list[[i]](value)
10: freduce(value, `_function_list`)
9: `_fseq`(`_lhs`)
8: eval(quote(`_fseq`(`_lhs`)), env, env)
7: eval(quote(`_fseq`(`_lhs`)), env, env)
6: withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
5: text %>% parse_transform_serialize_r(transformers, warn_empty = warn_empty) %>% 
       when(include_roxygen_examples ~ parse_transform_serialize_roxygen(., 
           transformers), ~.)
4: .f(.x[[i]], ...)
3: map(chunks$r_chunks, transformer_fun)
2: transform_mixed(context$contents, transformer, filetype = "Rmd")
1: styler:::style_active_file()
```

**Comment by lorenzwalthert** [comment]
Thanks. It’s most likely a bug introduced with #544. Can you please do binary search (e.g. remove second half of the file, try restyling, if it works, remove second half of remainder etc) to identify the code chunk where the error occurs and post the code here? Or create a reprex  from it?

**Comment by lorenzwalthert** [comment]
Are you using the CRAN release v1.2.0?

**Comment by IndrajeetPatil** [comment]
I am using the development version with the most recent commit. 

Okay, I will try to isolate the code chunk but it's gonna be a challenge since I am working with a pretty large `.Rmd` doc with many chunks.

**Comment by lorenzwalthert** [comment]
If it is too much work, you can post tue whole file I’d you don’t mind disclosing it. Or send it to my email you can find in the DESCRIPTION file of the style that package.

**Comment by IndrajeetPatil** [comment]
Oh, sure. Here it is: https://github.com/IndrajeetPatil/ggstatsplot/blob/master/vignettes/web_only/ggcoefstats.Rmd

**Comment by lorenzwalthert** [comment]
Implemented in master now. Note that for such big R Markdown files, you also benefit a lot from the caching feature implemented in v1.2.0 because every code chunk is styled and cached separately.

**Comment by IndrajeetPatil** [comment]
This is awesome, thanks!!
